opencensus/metrics/export/gauge.py

```diff
@@ -0,0 +1,222 @@
+# limitations under the License.
+
+from collections import OrderedDict
+import six
+import threading
+
+from opencensus.metrics.export import metric
+from opencensus.metrics.export import metric_descriptor
+from opencensus.metrics.export import point as point_module
+from opencensus.metrics.export import time_series
+from opencensus.metrics.export import value as value_module
+
+
+class GaugePointLong(object):
+    """An instantaneous measurement from a LongGauge.
+
+    A GaugePointLong represents the most recent measurement from a
+    :class:`LongGauge` for a given set of label values.
+    """
+
+    def __init__(self):
+        self.value = 0
+        self._value_lock = threading.Lock()
+
+    def __repr__(self):
+        return ("{}({})"
+                .format(
+                    type(self).__name__,
+                    self.value
+                ))
+
+    def add(self, val):
+        """Add `val` to the current value.
+
+        :type val: int
+        :param val: Value to add.
+        """
+        if not isinstance(val, six.integer_types):
+            raise ValueError("GaugePointLong only supports integer types")
+        with self._value_lock:
+            self.value += val
+
+    def set(self, val):
+        """Set the current value to `val`.
+
+        :type val: int
+        :param val: Value to set.
+        """
+        if not isinstance(val, six.integer_types):
+            raise ValueError("GaugePointLong only supports integer types")
+        with self._value_lock:
+            self.value = val
+
+
+class GaugePointDouble(object):
+    """An instantaneous measurement from a DoubleGauge.
+
+    A GaugePointDouble represents the most recent measurement from a
+    :class:`DoubleGauge` for a given set of label values.
+    """
+
+    def __init__(self):
+        self.value = 0.0
+        self._value_lock = threading.Lock()
+
+    def __repr__(self):
+        return ("{}({})"
+                .format(
+                    type(self).__name__,
+                    self.value
+                ))
+
+    def add(self, val):
+        """Add `val` to the current value.
+
+        :type val: float
+        :param val: Value to add.
+        """
+        with self._value_lock:
+            self.value += val
+
+    def set(self, val):
+        """Set the current value to `val`.
+
+        :type val: float
+        :param val: Value to set.
+        """
+        with self._value_lock:
+            self.value = float(val)
+
+
+class Gauge(object):
+    """A set of instantaneous measurements of the same type.
+
+    End users should use :class:`LongGauge` or :class:`DoubleGauge` instead of
+    using this class directly.
+
+    The constructor arguments are used to create a
+    :class:`opencensus.metrics.export.metric_descriptor.MetricDescriptor` for
+    converted metrics. See that class for details.
+    """
+
+    def __init__(self, name, description, unit, label_keys):
```

Comment:
I think we should validate all input params like ```name```, ```label_keys``` etc.

Reply:
Just that they're not null or empty? We check `label_keys` in the `MetricDescriptor` constructor: https://github.com/census-instrumentation/opencensus-python/blob/88ad971ccf497a43f5c08e77f2817b14b931d003/opencensus/metrics/export/metric_descriptor.py#L133-L137